internal/cluster/cluster_operations.go

```diff
@@ -467,15 +467,57 @@ func getFlattenedHostSpecsForRefs(ctx context.Context, hostRefs []vcf.HostRefere
 	return flattenedHostSpecs, nil
 }
 
-func getFlattenedVdsSpecsForRefs(vdsSpecs []vcf.VdsSpec) []map[string]interface{} {
-	flattenedVdsSpecs := *new([]map[string]interface{})
-	// Since backend API returns objects in random order sort VDSSpec list to ensure
-	// import is reproducible
-	sort.SliceStable(vdsSpecs, func(i, j int) bool {
-		return vdsSpecs[i].Name < vdsSpecs[j].Name
-	})
-	for _, vdsSpec := range vdsSpecs {
-		flattenedVdsSpecs = append(flattenedVdsSpecs, network.FlattenVdsSpec(vdsSpec))
+func getFlattenedVdsSpecsForRefs(ctx context.Context, clusterId string, apiClient *client.VcfClient) ([]map[string]interface{}, error) {
```

Comment:
This function needs to be updated to reflect the recent changed to the SDK.

I'll add a suggestion shortly.

Reply:
This may work:

```go
func getFlattenedVdsSpecsForRefs(ctx context.Context, clusterId string, apiClient *vcf.ClientWithResponses) ([]map[string]interface{}, error) {
	vdsParams := &clusters.GetVdsesParams{
		ClusterID: clusterId,
		Context:   ctx,
	}

	vdsRes, err := apiClient.GetVdsesWithResponse(ctx, vdsParams)
	if err != nil {
		return nil, err
	}

	vdsResponse, vcfErr := api_client.GetResponseAs[vcf.Vds](vdsRes)
	if vcfErr != nil {
		api_client.LogError(vcfErr)
		return nil, errors.New(*vcfErr.Message)
	}

	flattenedVdsSpecs := make([]map[string]interface{}, len(vdsResponse.Payload))
	for i, vds := range vdsResponse.Payload {
		portGroupSpecs := make([]*models.PortgroupSpec, len(vds.PortGroups))
		for j, pg := range vds.PortGroups {
			portGroupSpecs[j] = &models.PortgroupSpec{
				Name:          pg.Name,
				TransportType: pg.TransportType,
				ActiveUplinks: pg.ActiveUplinks,
			}
		}

		niocSpecs := make([]*models.NiocBandwidthAllocationSpec, len(vds.NiocBandwidthAllocations))
		for k, nioc := range vds.NiocBandwidthAllocations {
			if nioc != nil {
				niocSpecs[k] = &models.NiocBandwidthAllocationSpec{
					Type: &nioc.Type,
					NiocTrafficResourceAllocation: &models.NiocTrafficResourceAllocation{
						SharesInfo: &models.SharesInfo{
							Shares: nioc.NiocTrafficResourceAllocation.SharesInfo.Shares,
							Level:  nioc.NiocTrafficResourceAllocation.SharesInfo.Level,
						},
					},
				}
			}
		}

		vdsSpec := &models.VdsSpec{
			Name:                         vds.Name,
			IsUsedByNSXT:                 vds.IsUsedByNSXT,
			PortGroupSpecs:               portGroupSpecs,
			NiocBandwidthAllocationSpecs: niocSpecs,
		}

		flattenedVdsSpecs[i] = network.FlattenVdsSpec(vdsSpec)
	}

	return flattenedVdsSpecs, nil
}
```